ota-plus-web/app/com/advancedtelematic/controllers/FeedController.scala

```diff
@@ -33,26 +33,58 @@ class FeedController @Inject()(val conf: Configuration,
       case "device" =>
         implicit val devicesReads = feedResourcesReads("device")
         deviceRegistryApi.recentDevices(namespace, limit).map(_ \ "values").map(_.as[Seq[FeedResource]])
+
       case "device_group" =>
         implicit val deviceGroupsReads = feedResourcesReads("device_group")
-        deviceRegistryApi.recentDeviceGroups(namespace, limit).map(_ \ "values").map(_.as[Seq[FeedResource]])
+        enrich(
```

Comment:
You could pass `resourceType` to `enrich` and create the implicit there. Also `resourceType` could be a symbol, not a string.

Reply:
Yeah that would be an option, but `enrich` already takes almost too many params IMO. Also not all the `case`s  call `enrich` and I like the symmetry 😄
About symbol, it's deprecated in 2.13 and seems will be removed in 2.14, so I guess we better start getting used to it...

Reply:
`enrich` would take the same number of arguments then, because you can remove the implicit param. I don't see the point of introducing an implicit which you only use in 1 call.

Only symbol literals will be removed, but I see your point. Yay Python!

Reply:
I can see how having many implicits can get ugly... I realised they can all be simplified to one implicit before the case matching (see last commit). (Well, all but the one for "software" case, that actually is of a different type 🙂)